gson/src/main/java/com/google/gson/JsonDeserializer.java

```diff
@@ -48,10 +48,10 @@
  * deserializer:</p>
  *
  * <pre>
- * class IdDeserializer implements JsonDeserializer&lt;Id&gt;() {
+ * class IdDeserializer implements JsonDeserializer&lt;Id&gt; {
  *   public Id deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
  *       throws JsonParseException {
- *     return new Id((Class)typeOfT, id.getValue());
+ *     return new Id((Class) typeOfT, json.getAsJsonPrimitive().getAsLong());
  *   }
  * </pre>
```

Comment:
Sorry, I just noticed the original code is also missing a closing `}`:
```suggestion
 * }
 * </pre>
```

Reply:
updated